### PR TITLE
sample shows how lineWidth relates to the coordinate at which it's be…

### PR DESCRIPTION
**Pull request by djslavik**
…ing drawn

image

### CODE REVIEW

File: content/Canvas/canvasrenderingcontext2d.jsdoc [hunk 1]
```diff
@@ -568,13 +568,37 @@ Draw the path with %%#fill|**fill()**%% or
   var canvas = document.getElementById('canvas');
```
Comment (nkronlage) [inline content/Canvas/canvasrenderingcontext2d.jsdoc:568]: I think it would make more sense to replace the lineWidth example instead of beginPath.

**Review by nkronlage — COMMENTED** [review]
Thanks for the example!

This is a bit specific to vertical/horizontal lines - it'd be a bit hard to apply this to paths that are at an angle or have any bends in it. I wonder if using a stroke() is the best approach for this type of drawing or would using fillRect() be better? Have thoughts why stroke() is better here?

File: content/Canvas/canvasrenderingcontext2d.jsdoc (hunk 1 above)
Comment (djslavik) [reply content/Canvas/canvasrenderingcontext2d.jsdoc:568]: ah, of course! *facepalm*

File: content/Canvas/canvasrenderingcontext2d.jsdoc [hunk 2]
```diff
@@ -568,13 +568,37 @@ Draw the path with %%#fill|**fill()**%% or
...
   var context = canvas.getContext('2d');
 
-  context.lineWidth = 20;
+  var lineWidth = 20;
+  var y = 80;
+  context.strokeStyle = 'black';
+  context.lineWidth = lineWidth;
 
+  // align-middle (default)
```
[the remark this answers is not in the document]
Comment (djslavik) [reply content/Canvas/canvasrenderingcontext2d.jsdoc:576]: like it!

File: content/Canvas/canvasrenderingcontext2d.jsdoc [hunk 3]
```diff
@@ -568,13 +568,37 @@ Draw the path with %%#fill|**fill()**%% or
...
   context.beginPath();
-  context.moveTo(20,20);
-  context.lineTo(60,60);
-  context.lineTo(100,20);
+  context.moveTo(20, y);
+  context.lineTo(80, y);
+  context.stroke();
+
+  // align-top
```
[the remark this answers is not in the document]
Comment (djslavik) [reply content/Canvas/canvasrenderingcontext2d.jsdoc:582]: yup yup

File: content/Canvas/canvasrenderingcontext2d.jsdoc [hunk 4]
```diff
@@ -568,13 +568,37 @@ Draw the path with %%#fill|**fill()**%% or
...
+  context.lineTo(80 + 100, y + lineWidth/2);
   context.stroke();
+
+  // align-bottom
+  context.beginPath();
+  context.moveTo(20 + 200, y - lineWidth/2);
+  context.lineTo(80 + 200, y - lineWidth/2);
+  context.stroke();
+
+  // y-coordinate
```
[the remark this answers is not in the document]
Comment (djslavik) [reply content/Canvas/canvasrenderingcontext2d.jsdoc:594]: yes

File: content/Canvas/canvasrenderingcontext2d.jsdoc [hunk 5]
```diff
@@ -568,13 +568,37 @@ Draw the path with %%#fill|**fill()**%% or
   var canvas = document.getElementById('canvas');
   var context = canvas.getContext('2d');
 
-  context.lineWidth = 20;
+  var lineWidth = 20;
+  var y = 80;
```
[the remark this answers is not in the document]
Comment (djslavik) [reply content/Canvas/canvasrenderingcontext2d.jsdoc:572]: sure

**Comment by djslavik** [comment]
I hear you on non-lines. fillRect() would work just fine, perhaps better even, as it requires one to explicitly take care of the coordinates. In my case, the code was already using paths to decorate a rectangle with an inner top and bottom hairline borders. Perplexingly, they looked fuzzy instead of sharp. Things finally made sense once the realization hit about the paths being drawn around their middle (reference) line.

Here's, then, what I think might do the job - to modify the original example for lineWdith to show, drawn on it, the reference line that is also labeled as such right on the canvas. There seems to be enough power of suggestion here :)

**Comment by djslavik** [comment]
Shmebbe smth like this? :)

image

**Comment by nkronlage** [comment]
Ooh, that's nice! Did you already code it up to get that photo?

Also, you could update the description to say the line is centered on the coordinates so half the width is on either side of the path.

**Comment by djslavik** [comment]
I'm thinking of creating a brand new pull request with the change + description update.
Or should I update this existing one? Seems messy given it's been botched and kind of rrelevant by now.

**Comment by nkronlage** [comment]
Whatever is easier. A new one sounds good. Can you close this or need me to do it?

**Comment by djslavik** [comment]
Ok!

**Pull request closed without merging**